src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx

```diff
@@ -394,14 +409,29 @@ function ReportActionCompose({
         ],
     );
 
+    const debouncedValidate = lodashDebounce(
+        (value: string) => {
+            const taskCommentMatch = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
+            setIsCreatingTaskComment(!!taskCommentMatch);
+            if (taskCommentMatch) {
+                const title = taskCommentMatch?.[3] ? taskCommentMatch[3].trim().replace(/\n/g, ' ') : '';
+                validateTaskTitleMaxLength(title);
```

Comment:
@hoangzinh sure, updated

Reply:
Not sure why `perf-tests` failed, it pass in local 
![image](https://github.com/user-attachments/assets/48082f72-25c1-4a2b-8534-97f5569a5928)

Reply:
forgot to set baseline, here is the result

![image](https://github.com/user-attachments/assets/94df30b9-4ad7-4fdf-bcce-fa7824090c42)

@hoangzinh do you know what that means? It seems our change is causing extra rendering, right?

Reply:
@hoangzinh I've tried these but the error keeps showing.

```tsx

    const debouncedValidate = lodashDebounce(
        useCallback(
            (value: string) => {
                const taskCommentMatch = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
                if (taskCommentMatch) {
                    const title = taskCommentMatch?.[3] ? taskCommentMatch[3].trim().replace(/\n/g, ' ') : '';
                    setHasExceededMaxCommentLength(false);
                    validateTaskTitleMaxLength(title);
                } else {
                    setHasExceededMaxTitleLength(false);
                    validateCommentMaxLength(value, {reportID});
                }
            },
// choose one dependencies
            // [hasExceededMaxCommentLength, hasExceededMaxTaskTitleLength],
            // [reportID, setHasExceededMaxCommentLength, setHasExceededMaxTitleLength, validateCommentMaxLength, validateTaskTitleMaxLength],
        ),
        CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME,
        {leading: true},
    );

```

Reply:
@hoangzinh updated. it works when following combination of useCallback and useMemo similar to previous `useHandleExceedMaxCommentLength`


```tsx

    const validateMaxLength = useCallback(
        (value: string) => {
            const taskCommentMatch = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
            if (taskCommentMatch) {
                const title = taskCommentMatch?.[3] ? taskCommentMatch[3].trim().replace(/\n/g, ' ') : '';
                setHasExceededMaxCommentLength(false);
                validateTaskTitleMaxLength(title);
            } else {
                setHasExceededMaxTitleLength(false);
                validateCommentMaxLength(value, {reportID});
            }
        },
        [setHasExceededMaxCommentLength, setHasExceededMaxTitleLength, validateTaskTitleMaxLength, validateCommentMaxLength, reportID],
    );

    const debouncedValidate = useMemo(() => lodashDebounce(validateMaxLength, CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME, {leading: true}), [validateMaxLength]);


```

Reply:
@hoangzinh actually I've used that, but it doesn't work as expected, not sure why, let me check again

Reply:
@hoangzinh it now even worse after wrapping with useCallback and useMemo

![image](https://github.com/user-attachments/assets/9be6f312-f7b7-4daf-b461-e607407bb264)

Do you find something wrong with this code?
```tsx

    const validateMaxLength = useCallback(
        (value: string) => {
            const taskCommentMatch = value.match(CONST.REGEX.TASK_TITLE_WITH_OPTONAL_SHORT_MENTION);
            if (taskCommentMatch) {
                const title = taskCommentMatch?.[3] ? taskCommentMatch[3].trim().replace(/\n/g, ' ') : '';
                setHasExceededMaxCommentLength(false);
                validateTaskTitleMaxLength(title);
            } else {
                setHasExceededMaxTitleLength(false);
                validateCommentMaxLength(value, {reportID});
            }
        },
        [setHasExceededMaxCommentLength, setHasExceededMaxTitleLength, validateTaskTitleMaxLength, validateCommentMaxLength, reportID],
    );


    const debouncedValidate = useMemo(() => useDebounce(validateMaxLength, CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME, {leading: true}), [validateMaxLength]);


```

Reply:
> But I found a bug, if we paste an exceeded max creating task comment length and send quickly, we can send that message.

@hoangzinh It's expected with a 1500 ms time. The issue also occurs with current 10,000 validation, but the debounce time is reached before the text load/freeze completes. Should we separate back the debounce time?

Reply:
@hoangzinh 

The issue persist even with the optional chain

![image](https://github.com/user-attachments/assets/ef9de92b-70ca-41a6-b5ea-25940a42faa0)


I think it's better to use `lodashDebounce` instead of debugging and fixing `useDebounce`, since that hook is not widely used in the codebase.

Reply:
@hoangzinh  I can confirm that the issue also occurred for `MAX_COMMENT_LENGTH`

Here's the video, I've replaced `MAX_COMMENT_LENGTH` to `100`

In the video, I toggle between 100 and 101 character and when it's 101 character but it's still green, we can submit it


https://github.com/user-attachments/assets/061ca0c9-7adb-4956-b79c-024e5c8b99d5



>  I think we can leave it there until we have a real problem with it. At the moment, BE still returns errors for those cases

@hoangzinh agree, user will rarely does it intentionally.

Reply:
@hoangzinh Just refresh the page after making the code change to see the error. Is it working correctly on your end?

Reply:
@hoangzinh Here's mine

https://github.com/user-attachments/assets/8e7af851-9eb3-434c-8dae-ebf448bf7e4c

Reply:
@hoangzinh thanks, here's the mentioned original issue using useDebounce. When refreshing, the validation not triggers


https://github.com/user-attachments/assets/77c25759-ab79-45f7-9828-2fe6a65d9186

Reply:
@hoangzinh We still utilize lodashDebounce. please let me know if you have other feedback

Reply:
@hoangzinh bump